examples/fairmotion_interface_utils.py

```diff
@@ -0,0 +1,342 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import time
+from dataclasses import dataclass
+from enum import Enum
+from typing import List, Optional
+
+import magnum as mn
+from fairmotion.data import amass
+from fairmotion.ops import motion as motion_ops
+
+from habitat_sim.logging import LoggingContext, logger
+
+LoggingContext.reinitialize_from_env()
+
+#### Constants ###
+ROOT, FIRST, LAST = 0, 0, -1
+
+
+class MType(Enum):
+    TRANSITIVE = 0
+    SCENIC = 1
+
+
+class Motions:
+    """
+    The Move class is collection of stats that will hold the different movement motions
+    for the character to use when following a path. The character is left-footed so that
+    is our reference for with step the motions assume first.
+    """
+
+    @dataclass
+    class MotionData:
+        """
+        A class intended to handle precomputations of utilities of the motion we want to
+        load into the character.
+        """
+
+        # transitive motion is necessary for scenic motion build
+        def __init__(
+            self,
+            motion_,
+            type_,
+            transitive_motion_=None,
+            scenic_direction_offset: int = 0,
+        ) -> None:
```

Comment:
Please add typing and parameter documentation wherever possible. 
For example: 
- what is "scenic_direction_offset" here? 
- what type is expected for "type_", "motion_", and "transitive_motion_"?

Reply:
so for `transitive_motion` it is of type Motions.MotionData. It is the type that the class is so pylance doesn't recognize it. Is that a problem?

Reply:
https://stackoverflow.com/questions/33533148/how-do-i-type-hint-a-method-with-the-type-of-the-enclosing-class